common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -317,6 +348,22 @@ void release(DefaultHandle<T> handle) {
             }
         }
 
+        void unguardedRelease(DefaultHandle<T> handle) {
+            handle.unguardedToAvailable();
+            Thread owner = this.owner;
+            if (owner != null && Thread.currentThread() == owner && batch.size() < chunkSize) {
+                accept(handle);
+            } else if (owner != null && owner.getState() == Thread.State.TERMINATED) {
+                this.owner = null;
+                pooledHandles = null;
+            } else {
+                MessagePassingQueue<DefaultHandle<T>> handles = pooledHandles;
+                if (handles != null) {
+                    handles.relaxedOffer(handle);
+                }
+            }
```

Comment:
This block is duplicated with `release()`. Can we extract it into a method? Alternatively, if you are worried about inline depth, we can pull the state-change calls up into the relevant `DefaultHandle` methods, and have `LocalPool.release` just do this bit here.

Reply:
I think is fine to reduce the amount of duplicated code here, the most costy things going on here are elsewhere i.e.  'relaxedOffer` perform a CAS!